source/providers/Pict-Provider-DynamicFormSolverBehaviors.js

```diff
@@ -308,12 +308,34 @@ class PictDynamicFormsSolverBehaviors extends libPictProvider
 			return false;
 		}
 
-		// if there's a parent element, use that instead and we can color the whole input container area
-		if (tmpElementSet[0].parentElement)
+		let tmpElement = tmpElementSet[0];
+
+		// if we passed a class target, find the closest element with that class and apply the color to it
+		// otherwise, just apply it to the input element itself
+		if (pClassTarget)
 		{
-			tmpElementSet = [ tmpElementSet[0].parentElement ];
+			const tmpClosestTarget = [];
```

Comment:
No reason for this to be an array. Can just be `let tmpMatchedElement;` and check if it's truthy below. Or even just abandon it and use the outer `tmpElement`. The only reason for a second variable is to do the failure log / abort below.